deepdoc/parser/pdf_parser.py

```
@@ -196,6 +196,28 @@ def _has_color(self, o):
+
+    @classmethod
+    def _ocr_can_represent(cls, text, min_coverage=0.8):
+        """True if the OCR recogniser's alphabet covers this text well enough to be worth OCRing."""
+        if not text:
+            return True
+        if cls._OCR_ALPHABET is None:
+            try:
+                res = os.path.join(get_project_base_directory(), "rag/res/deepdoc/ocr.res")
+                with open(res, encoding="utf-8") as f:
+                    cls._OCR_ALPHABET = set(f.read())
+            except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)